File: Week 02/id_158/242.有效的字母异位词.cs
/*
 * @lc app=leetcode.cn id=242 lang=csharp
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
public class Solution {
    ///暴力
    public bool IsAnagram(string s, string t) {
        char[] arrS=s.ToCharArray();
        char[] arrT = t.ToCharArray();
        Array.Sort(arrS);
        Array.Sort(arrT);
        return Enumerable.SequenceEqual(arrS, arrT);
    }
    /*
      计数器
     */
    public bool IsAnagram(string s, string t) {
            if (s.Length != t.Length) return false;
            int[] alphabet = new int[26];
            for (int i = 0; i < s.Length; i++)
                alphabet[s[i] - 'a']++;
            for (int i = 0; i < t.Length; i++)
                alphabet[t[i] - 'a']--;
            foreach (int i in alphabet)
                if (i != 0) return false;
            return true;
    }

}
// @lc code=end

File: Week 02/id_493/242.有效的字母异位词.js
/*
 * @lc app=leetcode.cn id=242 lang=javascript
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) {
        return false;
    }
    let numObj = {};
    for (let a of s) {
        if (numObj.hasOwnProperty(a)) {
            numObj[a] = numObj[a] + 1;
        } else {
            numObj[a] = 1;
        }
    }
    for (let a of t) {
        if (numObj.hasOwnProperty(a)) {
            numObj[a] = numObj[a] - 1;
            if (numObj[a] < 0) {
                return false;
            }
        } else {
            return false;
        }
    }
    return true;
};
// @lc code=end

